### PR TITLE
Adding Picker Component

### PR DESCRIPTION
**Pull request by MaxwellKendall**
Thank you for taking the time to contribute to this library.

### Check List Before Merging! (For the Author)
- [x] Story Book has been rebuilt (new build artifacts in `/docs`)
- [x] Components have been rebuilt (new build artifacts in `/dist`)
- [x] All new components have been exported in `/index.js` `(if applicable)`
- [x] Design Review Complete
- [x] Relevant Draft release notes have been updated describing these changes ([use this template](https://github.com/fedspendingtransparency/data-transparency-ui/blob/master/draft_release_notes.md) to document your changes) `(if applicable)`
- [x] Integration Status Updated on relevant documentation page `(if applicable)`

### CODE REVIEW

File: babel.config.js [hunk 1]
```diff
@@ -5,7 +5,9 @@ const presets = [
             targets: {
                 ie: "10",
                 node: "10"
-            }
+            },
+            useBuiltIns: "entry",
+            corejs: { version: 3, proposals: true }
```
Comment (MaxwellKendall) [inline babel.config.js]: allows us to use auto-prefixing

File: postcss.config.js [hunk 2]
```diff
@@ -1,9 +1,6 @@
 module.exports = {
     plugins: {
-      'autoprefixer': {},
-      'css-nano': {}
-    },
-    autoprefixer: {
-      browsers: ['last 2 versions', 'ie 8-11', 'Firefox > 20']
+        autoprefixer: {},
+        'css-nano': {}
     }
```
Comment (MaxwellKendall) [inline postcss.config.js]: appends the browser-specific prefixes

File: pull_request_template.md [hunk 3]
```diff
@@ -1,10 +1,12 @@
...
-[ ] All new components have been exported in `/index.js` `(if applicable)`
-[ ] Design Review Complete
-[ ] Relevant Draft release notes have been updated describing these changes ([use this template](https://github.com/fedspendingtransparency/data-transparency-ui/blob/master/draft_release_notes.md) to document your changes) `(if applicable)`
-[ ] Integration Status Updated on relevant documentation page `(if applicable)`
+Thank you for taking the time to contribute to this library.
 
+### Check List Before Merging! (For the Author)
+- [ ] Story Book has been rebuilt (new build artifacts in `/docs`)
+- [ ] Components have been rebuilt (new build artifacts in `/dist`)
+- [x] All new components have been exported in `/index.js` `(if applicable)`
```
Comment (ebdabbs) [inline pull_request_template.md]: Do we want this to default to checked?

Comment (MaxwellKendall) [reply pull_request_template.md]: No -- fixed it!

File: .storybook/stories/pagination.stories.mdx [hunk 4]
```diff
@@ -124,7 +124,9 @@ changeLimit(pageSize);
 <Props of={Pagination} />
 
 ## Integration Status
-## USASpending
+Answers the question, where is this component used?
```
Comment (jonhill13) [inline .storybook/stories/pagination.stories.mdx]: I am not sure I understand this comment?

File: components/Picker.jsx [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+        onClick: PropTypes.func
+    })),
+    borderType: PropTypes.oneOf(['none', 'bottom', 'full']),
+    dropdownDirection: PropTypes.oneOf(['left', 'right']),
+    borderColor: PropTypes.string,
+    isFixedWidth: PropTypes.bool,
+    children: PropTypes.node
+};
+
+const getBorderType = (borderType, borderColor) => {
```
Comment (jonhill13) [inline components/Picker.jsx]: Do we want to include this? Could we just use our styles?

File: components/Picker.jsx [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+        if (a === selectedOption) return -1;
+        if (b === selectedOption) return 1;
+        // then, sort alphabetically
+        if (a < b) return -1;
+        if (a > b) return 1;
+        return 0;
+    };
+
+    const handleSort = (a, b) => {
+        if (sortFn(a, b) === 'default') return defaultSort(a, b);
```
Comment (jonhill13) [inline components/Picker.jsx]: couldnt we just set the default to `sortFn = () => 'defaultSort(a, b)'` and just return the `sortFn`?

File: components/Picker.jsx (hunk 5 above)
Comment (MaxwellKendall) [reply components/Picker.jsx]: You can just use `borderType="none"` and apply whatever css to design your own border.

File: components/Picker.jsx (hunk 6 above)
Comment (MaxwellKendall) [reply components/Picker.jsx]: What an idea....

File: components/Picker.jsx [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+
+        handleSetDimensions();
+        document.addEventListener('click', closeMenu);
+
+        return () => {
+            document.removeEventListener('click', closeMenu);
+        };
+    }, [expanded]);
+
+    const handleOptionClick = (cb) => (e) => {
```
Comment (jonhill13) [inline components/Picker.jsx]: What is going on here?

File: .storybook/stories/picker.stories.mdx [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+<Props of={Picker} />
+
+
+## This is Designed to be a [Controlled Component](https://reactjs.org/docs/forms.html#controlled-components)
+Which means the `selectedOption` and `onClick` are defined inside a parent component; in this case, `PickerWrapper`.
+
+## SortFn Prop
+This prop is a function which allows for a custom sort; however, it is only invoked after the selectedOption has been sorted to the highest index. The current design of this component requires this logic!
+
```
Comment (jonhill13) [inline .storybook/stories/picker.stories.mdx]: Can we add a comment about the default icon.

File: components/Picker.jsx [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+import PropTypes from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+require('../styles/components/_picker.scss');
+
+const fontAwesomeIconId = "usa-dt-picker__button-icon--svg";
+
+const propTypes = {
+    sortFn: PropTypes.func,
+    icon: PropTypes.string,
```
Comment (jonhill13) [inline components/Picker.jsx]: prolly a little more flexibility if we allow users to pass own icons.

File: .storybook/misc.jsx [hunk 10]
```diff
@@ -75,3 +75,29 @@ export const QuarterPickerWrapper = (props) => {
       </div>
   );
 };
+
```
Comment (ebdabbs) [inline .storybook/misc.jsx]: This is fine for now, but I would think about breaking `misc.jsx` out into smaller files before it becomes massive as we keep adding components.

File: components/Picker.jsx [hunk 11]
```diff
@@ -0,0 +1,221 @@
...
+    altText: PropTypes.string,
+    iconColor: PropTypes.string,
```
Comment (ebdabbs) [inline components/Picker.jsx]: Some of these props could be achieved with css overrides instead (colors, border) - I would lean towards keeping the component simpler and bare-bones functionality.

File: .storybook/stories/picker.stories.mdx (hunk 8 above)
Comment (MaxwellKendall) [reply .storybook/stories/picker.stories.mdx]: I'm gonna just make icon an element.

File: components/Picker.jsx (hunk 6 above)
Comment (MaxwellKendall) [reply components/Picker.jsx]: wait, actually no because the default sort requires `selectedOption` to be inside the scope of the function.

Comment (MaxwellKendall) [reply components/Picker.jsx]: wait.... no you're right.

File: components/Picker.jsx [hunk 12]
```diff
@@ -0,0 +1,221 @@
...
+                top: styles.top,
+                right: `${dimensions.right}px`
+            };
+        }
+        return styles;
+    };
+
+    return (
+        <div id={id} className={`usa-dt-picker ${className}`} ref={pickerRef} style={{ backgroundColor }}>
+            {icon !== 'none' && (
```
Comment (jonhill13) [inline components/Picker.jsx]: we could handle these

```
<div id={id} className={`usa-dt-picker ${className}`} ref={pickerRef} style={{ backgroundColor }}>
{this.icon()}
<div className="usa-dt-picker__dropdown-container">
{this.children()}
<ul className={`usa-dt-picker__list ${expanded ? '' : 'hide'}`} style={getDropdownListStyles()}>
{this.options()}
```

File: components/Picker.jsx (hunk 11 above)
Comment (MaxwellKendall) [reply components/Picker.jsx]: Agreed -- removed all of em!

**Pull request merged**